crates/application/src/lib.rs

```diff
@@ -187,6 +213,31 @@ impl ApplicationManager {
         Ok(())
     }
 
+    fn link_release(
+        &self,
+        application_id: &calimero_primitives::application::ApplicationId,
+        dir: &camino::Utf8Path,
+        version: &semver::Version,
+        link_path: &camino::Utf8Path,
+    ) -> eyre::Result<()> {
+        let base_path = format!("{}/{}/{}", dir, application_id, version);
+        info! {"{}", base_path};
+        fs::create_dir_all(&base_path)?;
+
+        let file_path = format!("{}/binary.wasm", base_path);
+        info!("{}", file_path);
+        match symlink(link_path, &file_path) {
+            Ok(_) => {}
+            Err(err) => eyre::bail!("Symlinking failed: {}", err),
+        }
```

Comment:
this wants to be an `if let`

```suggestion
        if let Err(err) = symlink(link_path, &file_path) {
            eyre::bail!("Symlinking failed: {}", err);
        }
```